bindings/wasm/crate/src/lib.rs

```diff
@@ -90,7 +95,52 @@ pub fn create_inbound_session(
         .set(receiving_instance);
 
     match result {
-        Ok((_, plaintext)) => Ok(plaintext),
+        Ok((session_id, ciphertext_json)) => Ok(vec![
+            JsValue::from_str(&session_id),
+            JsValue::from_str(&ciphertext_json),
+        ]
+        .into_boxed_slice()),
+        Err(e) => Err(JsValue::from_str(&e.to_string())),
+    }
+}
+
+#[wasm_bindgen]
+pub fn encrypt_message(
+    sending_handle: &str,
+    session_id: &str,
+    message: &str,
+) -> Result<String, JsValue> {
+    let instances = INSTANCE_MAP.lock().unwrap();
+    let mut instance = instances
+        .get(sending_handle)
+        .ok_or("sending_handle not found")?
+        .take();
+
+    let result = instance.encrypt_message_serialized(session_id, message);
+    // Do we actually need to do this?
+    instances.get(sending_handle).unwrap().set(instance);
+
+    match result {
+        Ok(ciphertext_json) => Ok(ciphertext_json),
+        Err(e) => Err(JsValue::from_str(&e.to_string())),
+    }
```

Comment:
This seems fine to me, however there are more idomatic ways to use a Match statement.

1. Pretty sure that RHS `Ok()` is not needed. Values are implicitly `Ok`
2. result is an 'Result' which you are trying to map the error. You can use `result.map_error(|e| JsValue::from_str(&e.to_string()))`